src/test/clojure/xtdb/authn_test.clj

```diff
@@ -0,0 +1,222 @@
+(ns xtdb.authn-test
+  (:require [clojure.test :as t]
+            [xtdb.authn :as authn] 
+            [xtdb.node :as xtn]
+            [xtdb.test-util :as tu])
+  (:import [dasniko.testcontainers.keycloak KeycloakContainer]
+           [java.net URI]
+           [java.util Base64]
+           [org.keycloak.representations.idm ClientRepresentation CredentialRepresentation UserRepresentation RealmRepresentation]
+           [xtdb.api Authenticator]))
+
+(defonce ^KeycloakContainer container
+  (KeycloakContainer. "quay.io/keycloak/keycloak:26.0"))
+
+(defn seed! [^KeycloakContainer c & {:keys [access-token-lifespan]}]
+  (with-open [admin-client (.getKeycloakAdminClient c)]
```

Comment:
I would probably have written this in Kotlin, because it's all interop - np that it's in Clojure, i.e. dw about changing it now :)